docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,406 @@
+noisy_value = execute(circuit, noise_level=NOISE_LEVEL)
+
+NUM_TWIRLED_VARIANTS = 300
+twirled_circuits = generate_pauli_twirl_variants(circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+
+# Average results executed over twirled circuits
+from functools import partial
+from mitiq import Executor
+pt_vals = Executor(partial(execute, noise_level=NOISE_LEVEL)).evaluate(twirled_circuits)
+twirled_result = np.average(pt_vals)
+
+
+print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
+print(f"Error with twirling: {abs(ideal_value - twirled_result) :.3}")
+```
+
+```{caution}
+It is worth noting that Pauli twirling's goal is to tailor the noise from coherent to incoherent. 
+
+Depending on the noise strength, type of coherent noise etc. this transformation might not give better results after the Pauli twirled circuit is executed. See the plot in the next section for an example.
+```
+
+## Combining Pauli Twirling with ZNE
+
+```{code-cell} ipython3
+from mitiq.zne import execute_with_zne
+
+executor=partial(execute, noise_level=NOISE_LEVEL)
+zne_pt_vals = []
+
+for twirled_circuit in twirled_circuits:
+    zne_pt_vals.append(execute_with_zne(twirled_circuit, executor))
+
+mitigated_result = np.average(zne_pt_vals)
+
+print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
+print(f"Error with twirling: {abs(ideal_value - twirled_result) :.3}")
+print(f"Error with ZNE + PT: {abs(ideal_value - mitigated_result) :.3}")
+
+```
+Accordingly, depending on the noise strength, a combination of PT and ZNE do not work that well compared to just PT or ZNE. Thus, it is important to understand when combining a noisy tailoring technique with an error mitigation technique provides a significant advantage. 
+
+```{code-cell} ipython3
+
+import matplotlib.pyplot as plt
+
+# Plot error vs noise strength
+noise_strength = np.linspace(0.0, 1.0, 50)
+error_no_zne_no_twirling = []
+error_with_twirling = []
+error_with_twirling_and_zne = []
+zne_vals = []
+ideal_values = []
+NUM_TWIRLED_VARIANTS = 30
+
+for strength in noise_strength:
+    ideal_value = execute(circuit, noise_level=0.0)
+    ideal_values.append(ideal_value)
+    noisy_value = execute(circuit, noise_level=strength)
+    id_noisy_diff = abs(ideal_value-noisy_value)
+    error_no_zne_no_twirling.append(id_noisy_diff)
+
+    twirled_circuits = generate_pauli_twirl_variants(circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+    pt_vals = Executor(partial(execute, noise_level=strength)).evaluate(twirled_circuits)
+    twirled_result = np.average(pt_vals)
+    twirled_noisy_diff = abs(ideal_value-twirled_result)
+    error_with_twirling.append(twirled_noisy_diff)
+
+    executor=partial(execute, noise_level=strength)
+    zne_vals.append(abs(ideal_value - execute_with_zne(circuit, executor)))
+    
+    zne_pt_vals = []
+    for i in twirled_circuits:
+        zne_pt_vals.append(execute_with_zne(i, executor))
+    mitigated_twirled_result = np.average(zne_pt_vals)
```

Comment:
This is basically the same code as in the previous section. Since it's the core part of the tutorial, at least when it comes to code, it should be in a function on its own.

Reply:
I don't really get what you are trying to suggest here. Do you want me to create a function that compares the expectation values?

Reply:
What I am suggesting here is that you could define a routine to encapsulate the code for ZNE + PT. Something on the lines of:
```python
def zne_with_pt(circuit, executor, num_twirled_variants):
    zne_pt_vals = []
    for twirled_circuit in generate_pauli_twirl_variants(circuit, num_circuits=num_twirled_variants):
        zne_pt_vals.append(execute_with_zne(twirled_circuit, executor))
    return np.average(zne_pt_vals)
```
And you can use it both in L379 and L333. This would highlight the steps that are involved in composing the two techniques.

Reply:
Ah, ok. I get what you are trying to say now. 

Previously, I thought you wanted me to define a function to make it easy to get all the expectation values in the `for` loop before the plot.